### PR TITLE
Use isComponentGlyph check where possible

### PR DESCRIPTION
**Pull request by brookescarlett**
## ✍️ Proposed changes

🎟 _Jira ticket:_ [LG-2593](https://jira.mongodb.org/browse/LG-2593) & [LG-2596](https://jira.mongodb.org/browse/LG-2596)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Style change (purely visual updates; if this is the case, attach a screenshot below!)
- [x] Code style change (purely visual updates; if this is the case, attach a screenshot below!)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

## Feedback Levels

Please try to mark your comments with one of these levels

1. Praise (👏 🚀 🔥 🍾)
2. Opinions, Ideas, Suggestions (💡 👀)
3. Let’s discuss @ Code review (🤷 🤔 🧐 )
4. Doesn’t meet agreed upon standards (😬 ⚠️ ‼️)
5. Missing an edge case (🛑 🙅 ❌)
6. Missing core functionality (⛔️ ☣️ 🚧 🌋 💀)

(Note: Use ⌘-⌃-␣ to open the Emoji keyboard!)

### CODE REVIEW

File: packages/icon/src/types/LGGlyph.ts [hunk 1]
```diff
@@ -3,6 +3,7 @@ import * as SVGR from './SVGR';
 
 export interface ComponentProps extends SVGR.ComponentProps {
   size?: number | 'small' | 'default' | 'large' | 'xlarge';
+  ['data-testid']?: string;
```
Comment (shaneeza) [inline packages/icon/src/types/LGGlyph.ts:6]: Why do we need to include this?

Comment (brookescarlett) [reply packages/icon/src/types/LGGlyph.ts:6]: Was causing an issue here: https://github.com/mongodb/leafygreen-ui/pull/1512/files#diff-999a04b89f74f252924b0289efa25728ca194ac073530555f9abdbbfd3e904b3L123

Comment (shaneeza) [reply packages/icon/src/types/LGGlyph.ts:6]: the link seems to be broken, its doesn't show the line or file,

Comment (brookescarlett) [reply packages/icon/src/types/LGGlyph.ts:6]: SideNavGroup line 123!

**Review by shaneeza — APPROVED** [review]

**Pull request merged**